### PR TITLE
HW#8_Trupkina

### PR DESCRIPTION
**Pull request by SvetlanaTr**

### CODE REVIEW

File: HW8(MVC)/unit.test.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+//     describe('On click button Add: should add into list ', () => {
+
+//     });
+// });
+
+// describe('Testing Model... ', () => {
+//     describe(' ', () => {
+
+//     });
+// });
```
Comment (DmitryKorolDevPro) [inline HW8(MVC)/unit.test.js]: don't store dead code

File: HW8(MVC)/view.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+class View {
+    constructor() {
+        this.list = document.getElementById('list');
+        this.addButton = document.querySelector('.addBtn');
+        this.saveButton = document.querySelector('.saveBtn');
+
+        this.input = document.getElementById('toDoEl');
+
+    }
+    displayAddElement() {
```
Comment (DmitryKorolDevPro) [inline HW8(MVC)/view.js:27]: invalid name.
you add three different elements.

File: HW8(MVC)/view.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+        const closeButton = document.createElement('span');
+        const closeButtonText = document.createTextNode('X');
+        closeButton.className = 'close';
+        closeButton.appendChild(closeButtonText);
+
+        todoElement.appendChild(closeButton);
+        this.list.appendChild(todoElement);
+
+        this.input.value = '';
```
Comment (DmitryKorolDevPro) [inline HW8(MVC)/view.js]: not expected side effect

File: HW8(MVC)/view.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+class View {
+    constructor() {
+        this.list = document.getElementById('list');
+        this.addButton = document.querySelector('.addBtn');
+        this.saveButton = document.querySelector('.saveBtn');
+
+        this.input = document.getElementById('toDoEl');
```
Comment (DmitryKorolDevPro) [inline HW8(MVC)/view.js]: for better readability, it is better to first describe the properties explicitly

File: HW8(MVC)/package.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "hw8",
+  "version": "1.0.0",
+  "description": "jest",
+  "main": "index.js",
+  "dependencies": {},
+  "devDependencies": {
+    "jest": "^26.6.3"
```
Comment (DmitryKorolDevPro) [inline HW8(MVC)/package.json]: we discussed why it is better to freeze package versions

File: HW8(MVC)/model.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        if (this.shouldAddElement()) {
+            view.displayAddElement();
+        } else {
+            view.showAlert();
+        }
+    }
+    //should add filter by completed or no
+    movingToComplete() {
+
+    }
```
Comment (DmitryKorolDevPro) [inline HW8(MVC)/model.js]: why is it here? the comment is not obvious

File: HW8(MVC)/model.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    shouldAddElement() {
+        return view.input.value !== '';
+    }
+    addTodo() {
+        if (this.shouldAddElement()) {
+            view.displayAddElement();
+        } else {
+            view.showAlert();
+        }
+    }
```
Comment (DmitryKorolDevPro) [inline HW8(MVC)/model.js]: the model is responsible for managing business logic. the model is not driving the view!

File: HW8(MVC)/model.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+    //save todolist state so user can access it later
+    saveElement() {
+        localStorage.setItem('todoList', view.list.innerHTML);
+        alert('Your TODO Saved!');
+    }
+
+    shouldAddElement() {
+        return view.input.value !== '';
+    }
```
Comment (DmitryKorolDevPro) [inline HW8(MVC)/model.js]: shouldAddElement - what is it?

File: HW8(MVC)/model.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    loadTodo() {
+        if (localStorage.getItem('todoList')) {
+            view.list.innerHTML = localStorage.getItem('todoList');
+        }
+    }
+
+    //save todolist state so user can access it later
+    saveElement() {
+        localStorage.setItem('todoList', view.list.innerHTML);
+        alert('Your TODO Saved!');
```
Comment (DmitryKorolDevPro) [inline HW8(MVC)/model.js]: why do you need this alert?

File: HW8(MVC)/model.js [hunk 10]
```diff
@@ -0,0 +1,36 @@
+import { view } from "./view.js";
+
+class Model {
+    constructor() { }
+
+    //function to load todo if list is found in local storage.
+    loadTodo() {
+        if (localStorage.getItem('todoList')) {
+            view.list.innerHTML = localStorage.getItem('todoList');
```
Comment (DmitryKorolDevPro) [inline HW8(MVC)/model.js]: the model is not driving the view

File: HW8(MVC)/model.js (hunk 8 above)
Comment (SvetlanaTr) [reply HW8(MVC)/model.js]: I tried to prevent enter an empty string @DmitryKorolDevPro

**Pull request closed without merging**